### PR TITLE
react-datepicker: Add `holidays` prop

### PR DESCRIPTION
**Pull request by shimech**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <https://github.com/Hacker0x01/react-datepicker/commit/a6bb275b9769bc2e0641fc0253671d97fd881f63>

### CODE REVIEW

File: types/react-datepicker/index.d.ts
Comment (shimech) [inline types/react-datepicker/index.d.ts:89]: `| undefined` is probably not necessary because this prop has `?`.
However, I'm following the former contributes, since the other optional props generally have `| undefined`.

Comment (MysteryBlokHed) [reply types/react-datepicker/index.d.ts:89]: FYI, `| undefined` is typically included because of the `exactOptionalPropertyTypes` config option: <https://www.typescriptlang.org/tsconfig#exactOptionalPropertyTypes>

**Review by MysteryBlokHed — APPROVED** [review]
LGTM, but will need approval from a types owner before merging

**Review by navya9singh — APPROVED** [review]

**Comment by shimech** [comment]
Ready to merge

**Pull request merged**
